### ISSUE TITLE
Player can switch to self-creative mode while server is running in survival mode.

### ISSUE THREAD

**Issue by Zhwt** [GeyserMC/Geyser#304]
**Describe the bug**
Player can switch himself to creative mode during a survival game session. They can't take out creative items though, but they can fly for a short distance. If they float too long, they will be kick out for reason "fly is not enabled", but if they fly for several short distances, nothing will happen and only the console will get prompted "xxx moved wrongly".

**To Reproduce**

1. Join the game, make sure server is running under survival mode.
2. Click on pause button, and click "Settings".
3. Switch personal game mode to creative and resume game.
4. Try to fly for a short distance, if you're lucky enough, you can get to some place that can't be reached normally such as rooftop of spawn lobby.

**Expected behavior**
Player cannot switch himself to creative mode.

**Screenshots / Videos**
Maybe later.

**Server Version**
```
> version
[19:15:02 INFO]: This server is running Paper version git-Paper-161 (MC: 1.15.2) (Implementing API version 1.15.2-R0.1-SNAPSHOT)
[19:15:02 INFO]: Checking version, please wait...
[19:15:06 INFO]: Previous version: git-Paper-155 (MC: 1.15.2)
[19:15:06 INFO]: You are 6 version(s) behind
```

**Geyser Version**
Using inventory branch #8 bukkit build.

**Minecraft: Bedrock Edition Version**
1.14.30.2 Android xbox version, not the original one.

**Additional Context**
In creative mode, player can break some blocks that they have no privilege to break, but the block will "respawn" at the same place almost immediately.
And one player also reports that most time they will be "pull back" to where they were before if they try to go through the blocks they break, but sometime they can manage to break into other's house if they have a high ping, in other words, they can "noclip" into other's house they normally have no privilege to enter, however this isn't a problem since they still have no privilege to interact with things in the house, but when their ping comes back to a normal level, they can't get out of the house due to lack of privileges, I didn't managed to reproduce this, but the "short distance fly" hack did work.

**Comment by endevrr** [comment]
This has already been fixed with #228 please update Geyser

**Comment by DoctorMacc** [comment]
Closing this issue because of outdated Geyser build. If it still persists comment and I will reopen.
